numpy/random/mtrand/mtrand.pyx

```diff
@@ -4527,6 +4541,179 @@ cdef class RandomState:
         x.shape = tuple(final_shape)
         return x
 
+    def complex_normal(self, loc=0.0, gamma=1.0, relation=0.0, size=None):
```

Comment:
This seems unnecessarily restricted, the wikipedia entry uses a multivariate version where `loc` is a vector and `gamma` and `relation` are square arrays. In fact,  multivariate_complex reduces to multivariate normal with the correct definition of covariance. Even with this restricted form, why not call `multivariate_normal`. You could add broadcasting to the latter if that is what you need, although some reshaping would be needed to transform the multivariate output to complex.

Numpy has a complex type and using that would simplify `bivariate_normal`

Reply:
It is possible the arange the row and columns of the 2d covariance coming from `gamma` and `relation` so that it remains symmetric and positive semidefinite and the result can be viewed as complex.
```
In [11]: b = zeros((4,4))

In [12]: b[::2, ::2] = ones((2,2))

In [13]: b[::2, 1::2] = ones((2,2)) + 1

In [14]: b[1::2, ::2] = ones((2,2)) + 2

In [15]: b[1::2, 1::2] = ones((2,2)) + 3

In [16]: b
Out[16]: 
array([[ 1.,  2.,  1.,  2.],
       [ 3.,  4.,  3.,  4.],
       [ 1.,  2.,  1.,  2.],
       [ 3.,  4.,  3.,  4.]])
```
That is just a change of basis `p,T @ cov @ p`, where p is a permutation.

Reply:
I felt a full `multivariate_complex_normal` would be limiting in the sense that most expressed demand was for scalar complex RVs and going through a full MV implementation would be very slow since it would require a SVD unless there was a fast path similar to what is here now.  I used the idea of moving from `normal`->`complex_normal` as a natural step.